python/ray/serve/metric/sink.py

```diff
@@ -0,0 +1,139 @@
+from typing import Dict
+from collections import Counter, namedtuple
+
+import ray
+from ray.serve.metric.types import MetricType, MetricMetadata, MetricBatch
+from ray.serve.utils import _get_logger
+
+logger = _get_logger()
+
+
+def make_metric_namedtuple(metadata: MetricMetadata, record: MetricBatch):
+    fields = ["name", "type"]
+    fields += list(metadata.default_labels.keys())
+    fields += list(record.labels.keys())
+
+    merged_labels = metadata.default_labels.copy()
+    merged_labels.update(record.labels)
+
+    tuple_type = namedtuple(metadata.name, fields)
+    return tuple_type(name=metadata.name, type=metadata.type, **merged_labels)
+
+
+class BaseSink:
+    """A sink is the place to store or forward metrics to external services"""
+
+    def __init__(self):
+        self.metadata = dict()
+        self.recording = []
```

Comment:
This name doesn't really tell me what it holds out of context. Maybe `self.metric_batches`?